ast/annotations.go

```diff
@@ -705,7 +718,7 @@ func (as *AnnotationSet) Flatten() []*AnnotationsRef {
 // 2. The 'package' scope entry, if any
 // 3. Entries for the 'subpackages' scope, if any; ordered from the closest package path to the fartest. E.g.: 'do.re.mi', 'do.re', 'do'
 // The returned slice is guaranteed to always contain at least one entry, corresponding to the given rule.
-func (as *AnnotationSet) Chain(rule *Rule) []*AnnotationsRef {
+func (as *AnnotationSet) Chain(rule *Rule) AnnotationsRefSet {
```

Comment:
I wanted to differentiate the output of `Chain` and `Flatten` as these impy different ordering, and it only makes sense to do insertion sort by `Insert` on the output of `Flatten`, i.e. a `FlatAnnotationsRefSet`. 

To make things more clear, we could return e.g. a `ChainedAnnotationsRefSet` from `Chain`. I opted not to, because we don't do anything special with that, and I didn't want to inflate the number of types unnecessarily.